diffrax/misc/errors.py

```diff
@@ -57,7 +57,7 @@ def raises(_arg):
         raises((pred, index))
     else:
         msg = (
-            "`pred` must either be a `bool`, a NumPy array, a JAX array, or a "
-            "zero-argument callable that returns a `bool` or JAX array."
+            f"`pred` {pred} must either be a `bool`, a JAX array, or a zero-argument callable "
+            f"that returns a `bool` or JAX array, instead of {type(pred)}."
```

Comment:
I'd don't hold to this as a hard-and-fast rule, but for something like this where an error here can almost only ever occur due to something totally crazy: I'd recommend putting the `f"{}"` values at the end of the docstring. That way if they're something with a huge `__str__` they won't blat the rest of the error message that comes after them.

Reply:
All good :)

Let me know if this looks better.